lib/src/builder.dart

```diff
@@ -62,19 +62,33 @@ import 'package:source_gen/source_gen.dart';
 /// will produce a "'.mocks.dart' file with such mocks. In this example,
 /// 'foo.mocks.dart' will be created.
 class MockBuilder implements Builder {
+  @override
+  final Map<String, List<String>> buildExtensions;
+
+  const MockBuilder(
+      {this.buildExtensions = const {
+        '.dart': ['.mocks.dart']
+      }});
+
   @override
   Future<void> build(BuildStep buildStep) async {
     if (!await buildStep.resolver.isLibrary(buildStep.inputId)) return;
     final entryLib = await buildStep.inputLibrary;
     final sourceLibIsNonNullable = entryLib.isNonNullableByDefault;
-    final mockLibraryAsset = buildStep.inputId.changeExtension('.mocks.dart');
+
+    if (buildStep.allowedOutputs.length > 1) {
+      throw ArgumentError('Build_extensions has conflicting outputs on file '
+          '`${buildStep.inputId.path}`, it usually caused by missconfiguration '
+          'on your `build.yaml` file');
+    }
+    final mockLibraryAsset = buildStep.allowedOutputs.single;
```

Comment:
[nit] The `single` could fail if the list was empty. How about

```suggestion
    final mockLibraryAsset = buildStep.allowedOutputs.singleOrNull;
    if (mockLibraryAsset == null) {
      throw ArgumentError('Build_extensions has missing or conflicting outputs for '
          '`${buildStep.inputId.path}`, this is usually caused by a misconfigured '
          'build extension override in `build.yaml`');
    }
```

Reply:
Nice catch!